app/src/pages/review.tsx

```diff
@@ -1,180 +1,151 @@
+      fetch(screenerUrl, {
+        method: 'POST',
+        body: JSON.stringify(body),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      })
+        .then((response) => {
+          return response.json()
+        })
+        .then((data: EligibilityScreenerResponse) => {
+          if (data.success) {
+            // Mark this session as submitted so duplicate entries won't be created.
+            setSession({ ...session, submitted: true })
+            // Route to the next page.
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            router.push(forwardRoute)
+          } else {
+            console.log(
+              'An error occurred while attempting to submit the session',
+              data.error
+            )
+            setErrorMessage(`${t('apiError')} Error: ${data.error}`)
+            // Reload current page.
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            router.push('')
+          }
+        })
+        .catch((error) => {
+          console.log(
+            'An error occurred while attempting to call fetch()',
+            error
+          )
+          setErrorMessage(`${t('apiError')}`)
+          // Reload current page.
+          // eslint-disable-next-line @typescript-eslint/no-floating-promises
+          router.push('')
+        })
+    }
+  }
 
   return (
     <>
-      <BackLink href="/contact" />
+      {errorMessage && <PageError alertBody={errorMessage} />}
+      <BackLink href={backRoute} />
       <h1>
         <Trans i18nKey="Review.title" />
       </h1>
       <p>
         <Trans i18nKey="Review.subHeader" />
       </p>
-      <ReviewCollection
-        headerKey="Review.eligibilityTitle"
-        editable={true}
-        editHref="/eligibility"
-        reviewElements={formatEligibilityResponses(session)}
-      />
-      <ReviewCollection
-        headerKey="Clinic.title"
-        editable={true}
-        editHref="/clinic"
-        reviewElements={formatClinicResponses(session)}
-      />
-      <ReviewCollection
-        headerKey="Contact.title"
-        editable={true}
-        editHref="/contact"
-        reviewElements={formatContactResponses(session)}
-      />
-      <ButtonLink href="/confirmation" labelKey="Review.button" />
+      <ReviewSection editable={true} session={form} />
+      <Button labelKey="Review.button" onClick={handleClick} />
     </>
   )
 }
 
-export const getServerSideProps: GetServerSideProps = async ({
-  locale,
-  req,
-}) => {
-  const prevRouteIndex = req.headers.referer?.lastIndexOf('/')
-  const previousRoute =
-    prevRouteIndex && req.headers.referer?.substring(prevRouteIndex)
-  let returnval: GetServerSidePropsResult<{ [key: string]: object }> = {
+export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
+  const baseUrl = process.env.BASE_URL || ''
```

Comment:
Notes for reviewers: I needed to pass in a `baseUrl` because my jest tests were failing with a complaint that `fetch()` can only handle absolute urls. That's actually not true for client side next.js, which can [totally handle relative routes](https://nextjs.org/docs/basic-features/data-fetching/client-side), but jest was unhappy, so I added this in to appease it.